### ISSUE TITLE
How can I set stimulus in stochastic model? Because stimulus, background, preflash and so on can not be variables in IQMstochsim2 but they can in IQMsimulate.

### ISSUE THREAD

**Issue by larowrow** [CBeelen/Phototransduction#1]

**Comment by CBeelen** [comment]
Yes, you're right - the stochastic simulation framework does not allow variables, which is why we cannot use the stimulus, background and preflash description like in `IQMsimulate`.
What I did for single photon responses was to set the initial condition in the model to 1 activated Rhodopsin - like that, you start out with an activated Rhodopsin at time=0 and simulate the single photon response from then on. For multiple photon responses, I simulated more than 1 single photon response and added up the PDE traces to feed them into the deterministic backend.
An example of that is this script, where I simulated single photon responses (with and without precoupling) by using models with two different initial conditions:
[Script](https://github.com/CBeelen/Phototransduction/blob/main/Scripts_cluster/cluster_rand.m)
Another easy way to do this would be to modify the initial conditions, IQMtools has a function for that, if I remember correctly.
If you wanted to define different, more complex stimuli, you can hard-code your stimulus into the model's differential equations - like that you cannot as easily specify different conditions for each simulation, but you could modify your model according to your parameters each time. For that you would have to define a rate of conversion from inactive to activated Rhodopsin, according to your stimulus.
I'll leave the issue open for now, feel free to reach out here if you have more questions on this!

**Comment by larowrow** [comment]
You mean that if I want to define more complex stimuli in stochastic simulation framework, such as natural light, I need to change parameters or numbers of matters in reaction when I use IQMtool, maybe every time point. There is no more simple method to do this. 

If the answer about above questin is right, as for  more complex stimuli, coding by myself instead of using IQMtool is a better way either in the random or deterministic framework.

**Comment by 978539023** [comment]
You mean that if I want to define more complex stimuli in stochastic simulation framework, such as natural light, I need to change parameters or numbers of matters in reaction when I use IQMtool, maybe every time point. There is no more simple method to do this. 

If the answer about above questin is right, as for  more complex stimuli, coding by myself instead of using IQMtool is a better way either in the random or deterministic framework.

**Comment by 978539023** [comment]
You mean that if I want to define more complex stimuli in stochastic simulation framework, such as natural light, I need to change parameters or numbers of matters in reaction when I use IQMtool, maybe every time point. There is no more simple method to do this. 

If the answer about above questin is right, as for  more complex stimuli, coding by myself instead of using IQMtool is a better way either in the random or deterministic framework.

**Comment by larowrow** [comment]
Why you didn’t do the same light adaptation study in stochastic simulation framework?
